### ISSUE TITLE
Fixed backgrounds - iPad 10.5 landscape orientation

### ISSUE THREAD

**Issue by jdub850000** [elementor/elementor#10951]
Hi,
- [x ] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [ x] The issue still exists against the latest stable version of Elementor.

## Description

When using fixed background elements and viewing on an iPad in portrait, images look sharp and behave acceptably, but once rotated to landscape, zoom in horribly to the point of artpixelation. 

This issue appears to have occured during the recent spate of updates as it was okay a couple of months ago. I have a couple of websites that seem to exhibit this issue - I need to fix it. I have found it also occurs with one of your own theme kits - Photography.

I have also found that it occurs with your newest theme release, so I won't paste a system info because you should be able to easily reproduce.

Landscape orientation is difficult to test properly during production due to the limited number of breakpoints - landscape orientation is sorely needed. Using Safari responsive mode didn't show the issue and in fact even showed the fixed backround with scroll intact.

## Isolating the problem

- [ x] This bug happens with only Elementor plugin active (and Elementor Pro).
- [ x] This bug happens with a default WordPress theme active.
- [x ] I can reproduce this bug consistently using the steps above.

## Environment

You can reproduce the problem by looking at your own Photgraphy theme at: https://library.elementor.com/photography-portfolio/ on an iPad Pro 10.5 (for example) in landscape mode. Elementor 2.9.6 and Elementor Pro 2.9.1.

I have tried disabling Elementor Pro and then reverting the version of Elementor from 2.9.6 back to 2.8.5 - and it has not made a difference.

<summary>System Info</summary>
Not applicable as it also happens with one of your own websites. The only other problem could be my iPad rendering, but I wouldn't have thought so as I have no problems elsewhere - It's running the latest iOS version.

```
</details>

**Comment by jdub850000** [comment]
Zero feedback from anyone on this - Is it not reproducible?

**Comment by jdub850000** [comment]
Still nothing back on this. I received two emails from Elementor saying they were still working and supporting us during these uncertain times, as a lot of internet related companies are doing - is this no longer the case?

**Comment by Netzlichter** [comment]
I have the exact same issue. Fixed backgrounds on my iPad 12,9" in landscape mode looking horrible! We really need a fix for this...
https://stackoverflow.com/questions/24154666/background-size-cover-not-working-on-ios
https://stackoverflow.com/questions/23236158/how-to-replicate-background-attachment-fixed-on-ios

https://wordpress.org/support/topic/ipad-tablet-landscape-mode-background-fixed-images-2-of-3-not-showing/
https://wordpress.org/support/topic/ipad-tablet-landscape-mode-background-picture-not-working/

Make it from fixed to small on iPad would do the trick. Like it is in portrait mode.

**Comment by zackmcr** [comment]
Same issue. Bump.

**Comment by Maxthesmall** [comment]
Did you try this fix?
https://github.com/elementor/elementor/issues/12656

**Comment by Maxthesmall** [comment]
It is ridicules that @elementorbot does nothing about this issue!!

**Comment by Janportfolio** [comment]
The problem is still there and after a long search, I found the solution, but it is outside Elementor.
My client site is designed with Elementor pro, Hello theme, child theme, canvas, or full-width elementor template.
I have used the BG image setting and wanted a slider function but the fixed option is only for one image and not for the slider BG. To get it running I skipped the slider option and used it to show the BG image without distortion to place it in the Background overlay but on the Ipad Pro in the landscape, the image is still too big and the pixelation is horrible. And no Image slider is available there. So after a long search, I have found a BG plugin. It's free to use and can be downloaded here: https://wordpress.org/plugins/simple-full-screen-background-image/  But this is only a static image background and it will show on every page or post (only global). I have upgraded the plugin to the pro version, $18,- a year (one site), and it is working great and also have the slider back. No scaling problems anymore on any device. You can find the pro version here: https://amplifyplugins.com/downloads/full-screen-background-images-pro/   Indeed, it's a workaround but I hope the Elementor theme will solve this in future versions of Elementor. For now I am happy I have found what I was looking for and hope this also could work for you.

**Comment by canyon1991** [comment]
I've encountered the same issue with the background image behavior while using `background-attachment: fixed` on different devices. Here's a summary of my findings:

**Desktop (MacOS - Edge & Safari):**
- Works as expected. The image is fixed, sharp, and displays correctly.

**Mobile (iOS - Edge & Safari) & Tablet Portrait (iPadOS - Edge & Safari) :**
- Behaves as if it is set to `background-attachment: scroll`. Acceptable, and the image still looks good.

**Tablet Landscape (iPadOS - Edge & Safari):**
- The background looks pixelated, and from what I could find out the image gets sized to the full size of the body/page instead of the viewport/screen.

---

**Workaround:**
To address the tablet landscape issue, I applied the following workaround:

1. Added a CSS Class to all containers that use `background-attachment: fixed` - Go to the target container > Advanced > CSS Classes. I used rollover as the name of my class.

2. Added a following custom CSS:

```
@media (pointer: coarse) {
	.rollover{
		background-attachment: scroll !important;
	} 
}
```

**Break down:**

`@media (pointer: coarse)`: This is a media query that targets devices with a coarse input method. The (pointer: coarse) condition is used to check if the device has a coarse input device, such as a touchscreen. It's a way to apply specific styles to devices with touchscreens.

`.rollover { background-attachment: scroll !important; }`: The rule sets the background-attachment property to scroll, and the !important declaration is used to override any conflicting styles that might exist elsewhere in the stylesheet.

In summary, this CSS code is saying that if the device has a coarse pointer (like a touchscreen), elements with the class `.rollover` should have their `background-attachment` set to `scroll`. 

---

Feel free to adjust this draft according to your preferences or provide more details.

**Comment by JPOak** [comment]
@canyon1991 Thanks for this. I tried various techniques to target Ipad Pro Landscape. Was having issues. I think this might be the best solution. I guess the drawback would be large devices like the Surface that has touch, but I can live with that.

**Comment by WiRiaa** [comment]
I‘m trying to fix the background but this It doesn‘t work for me as well. I‘m adding the code to the custom css but nothing happens, what could be the reason? There is an exclamation mark next to 'important' which is saying ,,use of important'> Could you please let me know how to solve this?

**Comment by canyon1991** [comment]
> I‘m trying to fix the background but this It doesn‘t work for me as well. I‘m adding the code to the custom css but nothing happens, what could be the reason? There is an exclamation mark next to 'important' which is saying ,,use of important'> Could you please let me know how to solve this?

If you trying to use my method, have you assigned the class to that element?  Can you provide a screenshot of the element that you are trying to use this on? 

The message you are seeing is a normal, use of "!important" is not best practice but it is required in situations like this where we don't have access to the main CSS and we are adding on top of it. Without it, you can't override the standard CSS.

**Comment by WiRiaa** [comment]
Thank you for the response, still I don‘t know what could be wrong, mabe you can see it on the screenshots:
<img width="1440" alt="Bildschirmfoto 2024-04-27 um 22 39 31" src="https://github.com/elementor/elementor/assets/168214214/8d94662d-9321-4e3b-9ed2-a64f7412e2a9">
 
<img width="1440" alt="Bildschirmfoto 2024-04-27 um 22 39 28" src="https://github.com/elementor/elementor/assets/168214214/19d1bafe-37ab-4017-9de9-324cf5a817e3">

**Comment by canyon1991** [comment]
> Thank you for the response, still I don‘t know what could be wrong, mabe you can see it on the screenshots: >

I will need to check it later with more time. However, my first thought would be to move the CSS code out of the element, "Custom CSS" and move it to the Page Custom CSS.

**Comment by anencephalic** [comment]
I've added this in and placed CSS in both container and page levels independently but it does not work.  Any other ideas?

**Comment by anencephalic** [comment]
Yes. I tried this and it did not work. On Aug 2, 2024, at 10:32, canyon1991 ***@***.***> wrote:﻿

I've added this in and placed CSS in both container and page levels independently but it does not work. Any other ideas?

I have updated my post above - #10951 (comment)
Let me know if that helps.

**Comment by JennJe** [comment]
> I've encountered the same issue with the background image behavior while using `background-attachment: fixed` on different devices. Here's a summary of my findings:
> 
> **Desktop (MacOS - Edge & Safari):**
> 
>     * Works as expected. The image is fixed, sharp, and displays correctly.
> 
> 
> **Mobile (iOS - Edge & Safari) & Tablet Portrait (iPadOS - Edge & Safari) :**
> 
>     * Behaves as if it is set to `background-attachment: scroll`. Acceptable, and the image still looks good.
> 
> 
> **Tablet Landscape (iPadOS - Edge & Safari):**
> 
>     * The background looks pixelated, and from what I could find out the image gets sized to the full size of the body/page instead of the viewport/screen.
> 
> 
> **Workaround:** To address the tablet landscape issue, I applied the following workaround:
> 
>     1. Added a CSS Class to all containers that use `background-attachment: fixed` - Go to the target container > Advanced > CSS Classes. I used rollover as the name of my class.
> 
>     2. Added a following custom CSS:
> 
> 
> ```
> @media (pointer: coarse) {
> 	.rollover{
> 		background-attachment: scroll !important;
> 	} 
> }
> ```
> 
> **Break down:**
> 
> `@media (pointer: coarse)`: This is a media query that targets devices with a coarse input method. The (pointer: coarse) condition is used to check if the device has a coarse input device, such as a touchscreen. It's a way to apply specific styles to devices with touchscreens.
> 
> `.rollover { background-attachment: scroll !important; }`: The rule sets the background-attachment property to scroll, and the !important declaration is used to override any conflicting styles that might exist elsewhere in the stylesheet.
> 
> In summary, this CSS code is saying that if the device has a coarse pointer (like a touchscreen), elements with the class `.rollover` should have their `background-attachment` set to `scroll`.
> 
> UPDATE: 2/08/2024
> 
> Here are some screen shot to show my settings that worked for me - the ones I highlight in red are the important ones, all the others you can change as you see fit
> 
>     1. Set container to use `background-attachment: fixed`
> 
> 
> <img alt="image" width="295" src="https://private-user-images.githubusercontent.com/56959429/354644448-447a5b7c-aa69-4112-b454-0e640a7209ad.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ1MTYxNzAsIm5iZiI6MTcyNDUxNTg3MCwicGF0aCI6Ii81Njk1OTQyOS8zNTQ2NDQ0NDgtNDQ3YTViN2MtYWE2OS00MTEyLWI0NTQtMGU2NDBhNzIwOWFkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI0VDE2MTExMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdhNGY4Y2YyODBmOTJjYzNlM2Q5NjExNDcyZTgxMjE0Y2ZiOGUxMmQ2YzkzYWNmOTMzMTU2MTdiNDQwMzZkN2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.lHAQLWPhhmv-eLba2DXptYpwgVmLKvD52cwPQMpSy8Q">
> 
>     2. Set that same container to have a class "rollover"
> 
> 
> <img alt="image" width="294" src="https://private-user-images.githubusercontent.com/56959429/354644961-8b2e7ff1-dbce-490d-9e55-c9529adcf4a6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ1MTYxNzAsIm5iZiI6MTcyNDUxNTg3MCwicGF0aCI6Ii81Njk1OTQyOS8zNTQ2NDQ5NjEtOGIyZTdmZjEtZGJjZS00OTBkLTllNTUtYzk1MjlhZGNmNGE2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI0VDE2MTExMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNjOTg4OTJjN2IyMmYyNmZkNzI1OTczMzE1YWJjYWI1NjA4ZDBiYjJjZWMzMjBmNzQxYjZhYjM5NmNhYzhhMzcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.63QaK4ZWbFKMKbBGDTA-OFVy22eKgWDpdY1KwBCGBoY">
> 
>     3. Write CSS for the Class on the Theme Customiser (I use the Hello Elementor Theme)
> 
> 
> <img alt="image" width="297" src="https://private-user-images.githubusercontent.com/56959429/354645729-0bbf38c1-9066-4cc8-bdc5-4751d6aba425.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ1MTYxNzAsIm5iZiI6MTcyNDUxNTg3MCwicGF0aCI6Ii81Njk1OTQyOS8zNTQ2NDU3MjktMGJiZjM4YzEtOTA2Ni00Y2M4LWJkYzUtNDc1MWQ2YWJhNDI1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI0VDE2MTExMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTExMGVkZWJlYzdmMDBiMmJlMGZmMTc1NzZkZGVhZjMzMGFlMWU4MzdkYzcxYWE1ZTYzNzYzZjEzOGY5ZmE3OGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.pG_K8PjkuluZ_aXkj2s0BQ1cnIxfVeb4DTfmPIHVlEk"> <img alt="image" width="299" src="https://private-user-images.githubusercontent.com/56959429/354646054-4342f99f-a513-4537-81ba-b8b4041d9f9b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ1MTYxNzAsIm5iZiI6MTcyNDUxNTg3MCwicGF0aCI6Ii81Njk1OTQyOS8zNTQ2NDYwNTQtNDM0MmY5OWYtYTUxMy00NTM3LTgxYmEtYjhiNDA0MWQ5ZjliLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI0VDE2MTExMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIyMmFjMmI0NjI2Nzk3ZGRlZjZiNWRlNzFlNzdmMGZhOTk5NWIyYzA0NWJhZWI3NmJkZjkzNDg2YzY5NTZiODgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-ALQJnlij_zlGIGMsQbDQkTs7sUiLZYUkuoF0yhSpDQ">
> 
> Hope this helps some people. You can always send me a DM and I will be happy to help.

OMG this is brilliant! The solution I have been seeking for years! Thank you so much!!!!

**Comment by JennJe** [comment]
> > I've added this in and placed CSS in both container and page levels independently but it does not work. Any other ideas?
> 
> I have updated my post above - [#10951 (comment)](https://github.com/elementor/elementor/issues/10951#issuecomment-1820007878)
> 
> Let me know if that helps.

OMG this is brilliant! Works like a charm! It's the solution I have been seeking for years! Thank you so much!!!!

**Comment by MariuszMilka** [comment]
Same problem here and the workaround works fine. However, since the query checks if the device has a touch screen and only them applies the change, how does it work on laptops with touch screen?

**Comment by canyon1991** [comment]
> Same problem here and the workaround works fine. However, since the query checks if the device has a touch screen and only them applies the change, how does it work on laptops with touch screen?

To be honest, I haven’t tested it myself, but I’d imagine it simply applies the “fix” as intended. One approach could be to tweak the @media query so it only targets specific screen sizes with coarse pointers—this would effectively exclude desktop and laptop devices.

That said, the issue you might run into is that some tablets may fall within the same resolution range defined in the @media query. As a result, the “fix” might not work as expected on those devices.

Personally, I’d rather some people didn’t see the “fancy” effect than others seeing a pixelated image.
